docs/content/preview/api/ysql/the-sql-language/statements/dml_insert.md

```diff
@@ -92,9 +120,34 @@ DO NOTHING | DO UPDATE SET *update_item* [ , ... ] [ WHERE *condition* ]
 
 #### *condition*
 
+An expression that returns a value of type boolean. 
+All rows will be locked when the ON CONFLICT DO UPDATE actio is taken, but only rows where this expression returns true will actually be updated.
+
+This condition is evaluated after a conflict has been identified as a candidate to update.
+
+Only `NOT DEFERRABLE` constraints and unique indexes are supported as arbiters.
+`exclusion` constraints are not supported as arbiters with `ON CONFLICT DO UPDATE`. 
+
+The `ON CONFLICT DO UPDATE` clause is a “deterministic” statement. 
+The command is not allowed to update any existing row more than once, a `cardinality violation` error will be raised in such cases.
+Rows to be inserted must not conflict with each other on the used constraint or arbiter index.
+
+`ON CONFLICT DO UPDATE` does not support updating the partition key of a conflicting row in a partitioned table that results in the row moving to a new partition.
+
+{{< note title="Tip" >}}
+
+It is preferable to use a unique index for `ON CONFLICT` instead of a constraint because you can replace the underlying unique index with another more or less equivalent index before dropping the index being replaced.
```

Comment:
```suggestion
It is recommended to use a unique index for `ON CONFLICT` instead of a constraint, because you can replace the underlying unique index with another more or less equivalent index before dropping the index being replaced.
```